ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryCleanupService.java

```diff
@@ -111,35 +114,55 @@ public String getAppId() {
       return appId;
     }
 
+    public String getQueryId() {
+      return queryId;
+    }
+
     @Override
     public void run() {
-      try {
-        final Path pathName = Paths.get(stateDir + "/" + appId);
-        final File directory = new File(String.valueOf(pathName.normalize()));
-        if (directory.exists()) {
-          FileUtils.deleteDirectory(directory);
-          LOG.warn("Deleted local state store for non-existing query {}. "
-                  + "This is not expected and was likely due to a "
-                  + "race condition when the query was dropped before.",
-              appId);
+      if (queryId.equals("")) {
+        try {
+          final Path pathName = Paths.get(stateDir + "/" + appId);
+          final File directory = new File(String.valueOf(pathName.normalize()));
+          if (directory.exists()) {
+            FileUtils.deleteDirectory(directory);
+            LOG.warn("Deleted local state store for non-existing query {}. "
+                    + "This is not expected and was likely due to a "
+                    + "race condition when the query was dropped before.",
+                appId);
+          }
+        } catch (Exception e) {
+          LOG.error("Error cleaning up state directory {}\n. {}", appId, e);
         }
-      } catch (Exception e) {
-        LOG.error("Error cleaning up state directory {}\n. {}", appId, e);
+        tryRun(
+            () -> SchemaRegistryUtil.cleanupInternalTopicSchemas(
+                appId,
+                serviceContext.getSchemaRegistryClient(),
+                isTransient),
+            "internal topic schemas"
+        );
+
+        tryRun(() -> serviceContext.getTopicClient().deleteInternalTopics(appId),
+            "internal topics");
+        tryRun(
+            () -> serviceContext
+                .getConsumerGroupClient()
+                .deleteConsumerGroups(ImmutableSet.of(appId)),
+            "internal consumer groups");
+      } else {
+        tryRun(
+            () -> SchemaRegistryUtil.cleanupInternalTopicSchemas(
+                appId + "-" + queryId ,
+                serviceContext.getSchemaRegistryClient(),
+                isTransient),
+            "internal topic schemas"
+        );
+        tryRun(
+            () -> serviceContext.getTopicClient().deleteInternalTopics(appId + "-" + queryId),
```

Comment:
Do we need to have separate branches like this, or could we just define a `queryTopicPrefix` String based on the `queryId` and then just plug that in to the original logic wherever it previously used `appId`? It seems like most of the logic should be the same, if not all of it

Similarly, I think we probably want to keep the part about checking if the directory still exists or not, if there's a race condition that can cause it to stick around in the current architecture then it probably exists for the shared runtime architecture as well. We should be able to do the same thing here, ie just define the `pathName` / `directory` based on whether it's a binpacked query or not, and then plug that in rather than branch.

FYI for NamedTopologies, instead of checking/deleting the whole state directory you would just need to get the NamedTopology directory, which is of the form <state_dir>/<application_id>/`__<queryId>__`

Reply:
I can probably pull that off. We don't want ot clean up the consumerGroups tho right?

Reply:
> We don't want ot clean up the consumerGroups tho right

Right